soccer/radio/SimRadio.cpp

```diff
@@ -24,11 +25,14 @@ SimRadio::SimRadio(Context* const context, bool blueTeam)
 
 bool SimRadio::isOpen() const { return _tx_socket.isValid(); }
 
-void SimRadio::send(Packet::RadioTx& packet) {
+void SimRadio::send(Packet::RadioTx& radioTx,
+                    const std::array<RobotIntent, Num_Shells>& intents,
+                    const std::array<MotionSetpoint, Num_Shells>& setpoints) {
+    construct_tx_proto(radioTx, intents, setpoints);
     grSim_Packet simPacket;
     grSim_Commands* simRobotCommands = simPacket.mutable_commands();
-    for (int i = 0; i < packet.robots_size(); i++) {
-        const Packet::Robot& robot = packet.robots(i);
+    for (int i = 0; i < radioTx.robots_size(); i++) {
+        const Packet::Robot& robot = radioTx.robots(i);
```

Comment:
When you test this, make sure that grSim handles the extra "empty" robots being sent nicely. Based on the [source](https://github.com/RoboCup-SSL/grSim/blob/master/src/sslworld.cpp#L549), it /should/ be fine, but its very easy to miss while testing if you don't sepcifcally try it. I think just settting a nonzero default command should make it obvious

Reply:
Sorry, I wasn't super clear. Also, I'm not 100% sure on this info, based on the code I think it worked this way.

I think before we would only create the `Robot` class when we got radio communication or saw it on camera. In sim, this would mean 6 `Robot` instantiations, each with their own control packet. Only those 6 would be sent to grSim. Based on my reading of the new code, 16 control packets would be sent, not just the 6 seen/activily controlled. The other 10 will be filled with zeros so it shouldn't cause any crashing on our end, but it's very possible that on their end, the extra 10 may cause some weird effects, including crashing, or controlling the oponent robots on accident.